resources/assets/js/components/MigrationForm.vue

```diff
@@ -0,0 +1,270 @@
+<template>
+    <div>
+        <div v-if="errors.length">
+            <div class="alert alert-danger">
+                <p v-for="error in errors">
+                    {{ error }}
+                </p>
+            </div>
+        </div>
+
+        <div class="row">
+            <div class="col-md-6">
+                <h3>Migration Settings</h3>
+
+                <div class="form-group">
+                    <input type="text" class="form-control" name="migration_name"
+                           placeholder="Type the name of your migration here" v-model="migration_name">
+                    <p class="form-text">Example: create users table, the file name and class naming conventions will be
+                        applied for you</p>
+                </div>
+                <div class="form-group">
+                    <input type="text" class="form-control" name="table_name" placeholder="MySQL table name"
+                           v-model="table_name">
+                </div>
+                <div class="form-group">
+                    <p>Migration Type:</p>
+
+                    <div class="form-check">
+                        <label class="form-check-label">
+                            <input type="radio" class="form-check-input" name="migration_type" value="create" v-model="migration_type" checked>
+                            Create (default) - if you want this migration to create a new database table
+                        </label>
+                    </div>
+                    <div class="form-check">
+                        <label class="form-check-label">
+                            <input type="radio" class="form-check-input" name="migration_type" value="modify" v-model="migration_type">
+                            Modify - choose this option to modify an existing table
+                        </label>
+                    </div>
+                </div>
+            </div>
+
+            <div class="col-md-6">
+                <h3>Columns</h3>
+                <div class="alert alert-info" v-if="columns.length === 0">
+                    Click the add column button below to get started
+                </div>
+                <div class="columns-list">
+                    <column
+                        v-for="(column, columnIndex) in columns"
+                        :column="column"
+                        :handleRemoveColumn="removeColumn.bind(this, columnIndex)"
```

Comment:
Props should be hyphenated.

Not keen on using `this` in templates. If this is just an event handler, use custom events.